library/ssl_tls.c

```diff
@@ -6451,8 +6543,11 @@ int mbedtls_ssl_config_defaults( mbedtls_ssl_config *conf,
 #endif /* MBEDTLS_SSL_PROTO_TLS1_3_EXPERIMENTAL */
 #endif
 
-#if defined(MBEDTLS_ECP_C)
+#if defined(MBEDTLS_ECP_C) && !defined(MBEDTLS_DEPRECATED_REMOVED)
             conf->curve_list = ssl_preset_suiteb_curves;
+            conf->supported_groups = NULL;
```

Comment:
Even if `DEPRECATED_REMOVED` is unset, shouldn't we use `conf->supported_groups` here? This way, we avoid the heap allocation at handshake time, and we can remove the preset curve lists altogether.

Reply:
Much nicer code! Amended